### PR TITLE
modules/monit: complete monit execution module functions

### PR DESCRIPTION
**Pull request by sbz**
add new functions configtest(), id(), reload(), version() and validate()
to the monit execution module

### CODE REVIEW

File: salt/modules/monit.py [hunk 1]
```diff
@@ -149,3 +151,105 @@ def status(svc_name=''):
...
+
+        salt '*' monit.version
+    '''
+    cmd = 'monit -V'
+    out = __salt__['cmd.run'](cmd).splitlines()
+    ret = out[0].split()
+    return ret[-1]
+
+
+def id(**kwargs):
```
Comment (cachedout) [inline salt/modules/monit.py]: We prefer not to have `**kwargs` in execution modules and instead pass arguments explicitely. Can this be rewritten?

Comment (sbz) [inline salt/modules/monit.py]: @cachedout Sure, I will do the update for the code and the doc string, you suggest to have ```def id(reset=False)``` instead ?

Comment (jfindlay) [inline salt/modules/monit.py]: @sbz, that would be great.  Also you may want to reconsider renaming `id` to `id_` and also have at the top of the file:
```py
__func_alias__ = {
    'id_': 'id',
}
```

**Comment by sbz** [comment]
@jfindlay Done, thank you for the suggestion

**Comment by cachedout** [comment]
Looks great! Thank you so much, @sbz

File: salt/modules/monit.py [hunk 2]
```diff
@@ -149,3 +158,109 @@ def status(svc_name=''):
...
+        id_pattern = re.compile(r'Monit id (?P<id>[^ ]+)')
+        cmd = 'echo y|monit -r'
+        out = __salt__['cmd.run_all'](cmd, python_shell=True)
+        ret = id_pattern.search(out['stdout']).group('id')
+        return ret if ret else False
+    else:
+        cmd = 'monit -i'
+        out = __salt__['cmd.run'](cmd)
+        ret = out.split(':')[-1].strip()
+    return ret
```
Comment (mcansky) [inline salt/modules/monit.py]: for consistency, you could have that return in the else or set ret to False if it's None in the if block and let that last line return what ever is in ret ...

**Comment by sbz** [comment]
@cachedout Thanks for the merge!

**Pull request merged**